solr/core/src/java/org/apache/solr/util/stats/MetricUtils.java

```diff
@@ -787,10 +787,16 @@ public static <T extends PlatformManagedObject> void addMXBeanMetrics(
                 }
               });
 
+      // if BeanInfo retrieval failed, return early
       if (beanInfo == null) {
         return;
       }
       for (final PropertyDescriptor desc : beanInfo.getPropertyDescriptors()) {
+        // skip properties without a read method
+        if (desc.getReadMethod() == null) {
```

Comment:
this is fine but I think it'd be nicer to save the result and use it later rather than double-lookup.  Also nicer to get the read method inside the try-finally in case there's some issue even calling getReadMethod